src/components/bannerCard.tsx

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import "./bannerCard.scss";
+
+import * as React from "react";
+
+import * as moment from "moment";
+
+import { Button } from "azure-devops-ui/Button";
+import { ButtonGroup } from "azure-devops-ui/ButtonGroup";
+import { Card } from "azure-devops-ui/Card";
+import { FormItem } from "azure-devops-ui/FormItem";
+import { IStatusProps, Status, Statuses, StatusSize } from "azure-devops-ui/Status";
+import { TextField, TextFieldStyle } from "azure-devops-ui/TextField";
+import { Tooltip } from "azure-devops-ui/TooltipEx";
+
+import { GlobalMessageBanner, Level } from "../models/global-message-banners";
+import { VstsService } from "../services/vsts-service";
+import { ExpiryPicker } from "./expiryPicker";
+import { LevelDropdown } from "./levelDropdown";
+
+interface IBannerCardProps {
+    banner: GlobalMessageBanner;
+    onSave(banner: GlobalMessageBanner): void;
+    onDelete(): void;
+}
+
+interface IBannerCardState extends Partial<GlobalMessageBanner> {
+    dirty: boolean;
+    loading: boolean;
+    expanded: boolean;
+    expiryErrorText?: string;
+    messageErrorText?: string;
+}
+
+export class BannerCard extends React.Component<IBannerCardProps, IBannerCardState> {
+    constructor(props: IBannerCardProps) {
+        super(props);
+
+        this.state = {
+            level: this.props.banner.level,
+            expirationDate: this.props.banner.expirationDate,
+            message: this.props.banner.message,
+            dirty: false,
+            loading: false,
+            expanded: false,
+        };
+    }
+
+    public componentDidMount(): void {
+        this.setupFields();
+    }
+
+    public componentDidUpdate(prevProps: IBannerCardProps): void {
+        if (JSON.stringify(prevProps.banner) !== JSON.stringify(this.props.banner)) {

```

Comment:
I probably asked this before but should we worry about performance when comparing the diff using JSON.stringify?

Reply:
It is the fastest method I have found to compare objects. Various hashing algorithms are significantly worse in perf.